File: update1.js
const fs = require('fs');

let lines = fs.readFileSync('src/app/components/CleaningMapClient.tsx', 'utf8').split('\n');

// 1. Add MapZoomTracker -> Add MapClickHandler
const start1 = lines.findIndex(line => line.includes('function MapZoomTracker'));
if (start1 !== -1) {
    lines.splice(start1 + 6, 0, 
`function MapClickHandler({ onMapClick }: { onMapClick: () => void }) {
    useMapEvents({
        click: () => onMapClick()
    });
    return null;
}`);
}

// 2. Add state
const start2 = lines.findIndex(line => line.includes('const [focusedGroup, setFocusedGroup]'));
if (start2 !== -1) {
    lines.splice(start2 + 1, 0, "    const [focusedTerritoryId, setFocusedTerritoryId] = useState<string | null>(null);");
}

// 3. Add MapClickHandler component instance
const start3 = lines.findIndex(line => line.includes('<MapZoomTracker onZoomChange={setCurrentZoom} />'));
if (start3 !== -1) {
    lines.splice(start3 + 1, 0, "                    <MapClickHandler onMapClick={() => setFocusedTerritoryId(null)} />");
}

fs.writeFileSync('src/app/components/CleaningMapClient.tsx', lines.join('\n'), 'utf8');
console.log("update 1 done");

File: update1.py
import json

with open("src/app/components/CleaningMapClient.tsx", "r", encoding="utf-8") as f:
    lines = f.readlines()

# 1. MapZoomTracker -> Add MapClickHandler
start1 = -1
for i, line in enumerate(lines):
    if "function MapZoomTracker" in line:
        start1 = i
        break

if start1 != -1:
    lines.insert(start1 + 6, """
function MapClickHandler({ onMapClick }: { onMapClick: () => void }) {
    useMapEvents({
        click: () => onMapClick()
    });
    return null;
}
""")

# 2. Add state
start2 = -1
for i, line in enumerate(lines):
    if "const [focusedGroup, setFocusedGroup]" in line:
        start2 = i
        break

if start2 != -1:
    lines.insert(start2 + 1, "    const [focusedTerritoryId, setFocusedTerritoryId] = useState<string | null>(null);\n")

# 3. Add MapClickHandler
start3 = -1
for i, line in enumerate(lines):
    if "<MapZoomTracker onZoomChange={setCurrentZoom} />" in line:
        start3 = i
        break

if start3 != -1:
    lines.insert(start3 + 1, "                    <MapClickHandler onMapClick={() => setFocusedTerritoryId(null)} />\n")

with open("src/app/components/CleaningMapClient.tsx", "w", encoding="utf-8") as f:
    f.writelines(lines)
